aten/src/ATen/native/quantized/cpu/qnnpack/src/q8gemm/4x8-aarch32-neon.S

```diff
@@ -78,18 +78,30 @@ BEGIN_FUNCTION pytorch_q8gemm_ukernel_4x8__aarch32_neon
     # Load w
     # - ip = w
     LDR ip, [sp, 4]
-    PUSH {r4, r5, r6, r7, r8}
+    PUSH {r4, r5, r6, r7, r8, r9}
 
-    VPUSH {d8-d15}
     # Load quantization params
     # - r7 = quantization_params
-    LDR r7, [sp, 104]
+    LDR r7, [sp, 44]
+    VPUSH {d8-d15}
+
+    # Load bias0123, bias4567
+    VLDM ip!, {d16-d19}
 
     # Load output channel index
-    LDR r5, [sp, 100]
+    LDR r5, [sp, 104]
+    CMP r0, 2
```

Comment:
Where are you using the result of this comparison?  I don't seem to be able to spot it.

Reply:
This is from the original code. It is just movement of code that seem to have hidden its usage. It here: https://github.com/pytorch/pytorch/pull/37626/files/20a19b900e54e205185d5ce54a3ec327e54e12c2#diff-f682e6802902a89ec92f72efe5ec2a5bR120

Reply:
This is very dangerous.  Condition flags can be implicitly set as a side effect by quite a few instructions which makes putting too much distance between condition checks and their actual usage dangerous.  For instance if someone a month from now modifies this code to add a SUBS in between this  CMP and CSEL this code will break.

Reply:
OK, so I think you could also say that if someone is changing this code then they probably understand what they are doing. I do understand if this code gets larger it become intractable but probably you wont write really long handwritten assembly. Having said all that, in this instance I dont think there is much justification to move the cmp around. I was trying to hide latency but I doubt it helps much. I did not really account the point you mentioned, so thats fair. I will move it back to where it was.

Reply:
Again, as mentioned before, all my comments are merely suggestions and you are more than welcome to do as you see fit.

Reply:
Made the changes.